src/frontend/web_application/src/components/MessageList/components/Message/index.jsx

```diff
@@ -15,48 +34,119 @@ class Message extends Component {
 
   constructor(props) {
     super(props);
-    this.state = { isActive: false };
-    this.handleHover = this.handleHover.bind(this);
-    this.handleBlur = this.handleBlur.bind(this);
+    this.state = {
+      body: '',
+      isExpanded: true,
+      isTooLong: false,
+      isActive: false,
+    };
+    this.handleActiveClick = this.handleActiveClick.bind(this);
+    this.handleExpandClick = this.handleExpandClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.setBodyHeight();
+  }
+
+  setBodyHeight() {
+    const message = this.props.message;
+    const body = message.body;
+    if (body.length > 140) {
+      this.setState({
+        isTooLong: true,
+        isExpanded: false,
+        body: body.substring(0, 140),
+      });
+    } else {
+      this.setState({
+        body,
+      });
+    }
   }
 
-  handleHover() {
-    this.setState({ isActive: true });
+  handleActiveClick() {
+    this.setState({
+      isActive: !this.state.isActive,
+    });
   }
 
-  handleBlur() {
-    this.setState({ isActive: false });
+  handleExpandClick() {
+    const message = this.props.message;
+    const body = message.body;
+    if (this.state.isExpanded) {
+      this.setState({
+        isExpanded: false,
+        body: body.substring(0, 140),
+      });
+    } else {
+      this.setState({
+        isExpanded: true,
+        body,
+      });
+    }
   }
 
   render() {
     const { message, __ } = this.props;
     const author = message.participants.find(participant => participant.type === 'from');
```

Comment:
Only for the cordova build, it seems

https://github.com/CaliOpen/Caliopen/blob/3fca4a85fc4508467a08d0a5c038290b650e01c8/src/frontend/web_application/webpack/webpack.config.cordova.js